cmd/prometheus-config-reloader/main.go

```diff
@@ -227,3 +253,91 @@ func createOrdinalEnvvar(fromName string) error {
 	val := reg.FindString(os.Getenv(fromName))
 	return os.Setenv(statefulsetOrdinalEnvvar, val)
 }
+
+type prometheusReloadTracker struct {
+	baseURL *url.URL
+	client  http.Client
+	logger  log.Logger
+
+	lastReload  time.Time
+	lastSuccess bool
+}
+
+func (prt *prometheusReloadTracker) getConfigStatus(ctx context.Context) (bool, time.Time, error) {
+	r, err := http.NewRequestWithContext(ctx, "GET", prt.baseURL.JoinPath("/api/v1/status/runtimeinfo").String(), nil)
+	if err != nil {
+		return false, time.Time{}, err
+	}
```

Comment:
Nice idea to get reload status even when using signal 👍

Reply:
Yes otherwise it would be like shooting in the dark. This was the blocking point for me to move this forward.